packages/typespec-ts/test/modularIntegration/generated/models/inheritance/enum-discriminator/src/rest/models.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/** Test extensible enum type for discriminator */
+export interface DogParent {
+  /** Weight of the dog */
+  weight: number;
+  kind: string;
+}
+
+/** Golden dog model */
+export interface Golden extends DogParent {
+  /** discriminator property */
+  kind: "golden";
+}
+
+/** Test fixed enum type for discriminator */
+export interface SnakeParent {
+  /** Length of the snake */
+  length: number;
+  kind: "cobra";
+}
+
+/** Cobra model */
+export interface Cobra extends SnakeParent {
+  /** discriminator property */
+  kind: "cobra";
+}
+
+/** Test extensible enum type for discriminator */
+export type Dog = DogParent | Golden;
```

Comment:
I know using 'Parent' is to be consistent with HLC, but this line is still a big strange.
Can we have some other options to define the name?